packages/neuron-wallet/src/services/hardware/ledger.ts

```diff
@@ -78,12 +78,25 @@ export default class Ledger extends Hardware {
       context = txs.map(i => rpc.paramsFormatter.toRawTransaction(i.transaction))
     }
 
-    const signature = await this.ledgerCKB!.signTransaction(
-      path === hd.AccountExtendedPublicKey.pathForReceiving(0) ? this.defaultPath : path,
-      rawTx,
-      witnesses,
-      context,
-      this.defaultPath
+    const hdPath = path === hd.AccountExtendedPublicKey.pathForReceiving(0) ? this.defaultPath : path
+    const signature = await this.ledgerCKB!.signTransaction(hdPath, rawTx, witnesses, context, this.defaultPath).catch(
+      error => {
+        const errorMessage = error instanceof Error ? error.message : String(error)
+        logger.error(
+          'Ledger: failed to sign the transaction ',
+          errorMessage,
+          ' HD path:',
+          hdPath,
+          ' raw transaction:',
+          JSON.stringify(rawTx),
```

Comment:
It might be sensitive to expose transactions in the log. Is it possible to mask some fields, e.g., `args` to be `abc****xyz` and `capacity` to be `0x1*****`(keep the char count consistent)

Reply:
If a transaction is intended to be sent, it becomes public. I don't think it matters to log a raw transaction in the log file

Reply:
The transaction goes public but the sender in reality is a secret. However, the transaction exposed in the log builds the connection between the transaction and the person.

Reply:
Maybe we can encrypt the sensitive info and decrypt it in private workspace.

Reply:
It's quite challenging to fully sanitize a log file to protect all user privacy, as information such as `args`, `data`, and `witnesses` can be linked to a user's public key(public key hash or lock script hash). If we fully or partially hide these parts, it is probably not going to help debug

IMO, the logged data only includes failed transactions, which doesn't reveal too much information. Moreover, it's more effective to track the user through transaction patterns rather than getting logs directly from the user. This is because regular UTxO transactions contain change addresses, and the location of the change UTxOs is relatively consistent

Reply:
> It's quite challenging to fully sanitize a log file to protect all user privacy, as information such as `args`, `data`, and `witnesses` can be linked to a user's public key(public key hash or lock script hash). If we fully or partially hide these parts, it is probably not going to help debug
> 
> IMO, the logged data only includes failed transactions, which doesn't reveal too much information. Moreover, it's more effective to track the user through transaction patterns rather than getting logs directly from the user. This is because regular UTxO transactions contain change addresses, and the location of the change UTxOs is relatively consistent

Failed transactions are supposed to be on the chain, so it reveals users' purpose completely, and in advance.

I'm not persuaded by the reason of `more effective to track the users` so the info can be public. As mentioned above, the info can be encrypted, by doing so, the info is still masked while the raw transaction is accessible to limited developers.

I treat the failed transaction as sensitive info because one key feature of Neuron is that the next transaction is unpredictable. If the failed transaction is exposed to the public, it becomes predictable.